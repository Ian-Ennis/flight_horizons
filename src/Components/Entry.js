import React from "react";

/* 
Initial entry screen animations render when user first loads page
(window.sessionStorage.getItem("firstLoadDone") === null)

Animations specified in index.css
(@keyframes entry-title-fade-in, entry-title-fade-out, 
entry-visual-fade-in, launch, entry-visual-fade-out)

Images stored in bucket in AWS S3
*/

function Entry() {
  return (
    <div id="entry_container" data-testid="entry_component">
      <div id="entry_title_container">
        <img
          id="entry_title_animated"
          src="https://flight-horizons.s3.us-west-2.amazonaws.com/title.png"
          alt="flight_horizons_title"
        />
      </div>
      <div id="entry_visual_container">
        <img
          id="entry_visual_animated"
          src="https://flight-horizons.s3.us-west-2.amazonaws.com/spaceship.png"
          alt="spaceship_animation"
        />
      </div>
    </div>
  );
}

export default Entry;
