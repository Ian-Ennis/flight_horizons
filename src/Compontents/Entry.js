import title from '../Images/title.png'
import spaceship from '../Images/spaceship.png'

function Entry() {

    return (
        <div id="entry_container">
            <div id="entry_title_container">
                {/* <p>flight horizons</p> */}
                <img id="entry_title_animated" src={title} alt="flight_horizons"/>
            </div>
            <div id="entry_visual_container">
                <img id="entry_visual_animated" src={spaceship} alt="entry_visual"/>
            </div>
        </div>
    )
}

export default Entry

