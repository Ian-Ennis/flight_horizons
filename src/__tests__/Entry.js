import { render } from '@testing-library/react'
// import { render, screen } from '@testing-library/react';
import Entry from "../Components/Entry"

it('matches the DOM snapshot', () => {
  const domTree = render(<Entry />)
  expect(domTree).toMatchSnapshot();
})

// it('it renders the entry page', () => {
//   render(<Entry />);

//   const appTitleImage = screen.getByAltText('flight_horizons_title')
//   expect(appTitleImage).toBeInTheDocument();
//   expect(appTitleImage).toHaveAttribute('src', 'https://flight-horizons.s3.us-west-2.amazonaws.com/title.png')

//   const spaceshipVisual = screen.getByAltText('spaceship_animation')
//   expect(spaceshipVisual).toBeInTheDocument();
//   expect(spaceshipVisual).toHaveAttribute('src', 'https://flight-horizons.s3.us-west-2.amazonaws.com/spaceship.png')
// });