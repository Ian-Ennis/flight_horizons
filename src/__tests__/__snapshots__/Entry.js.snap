// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the DOM snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        id="entry_container"
      >
        <div
          id="entry_title_container"
        >
          <img
            alt="flight_horizons_title"
            id="entry_title_animated"
            src="https://flight-horizons.s3.us-west-2.amazonaws.com/title.png"
          />
        </div>
        <div
          id="entry_visual_container"
        >
          <img
            alt="spaceship_animation"
            id="entry_visual_animated"
            src="https://flight-horizons.s3.us-west-2.amazonaws.com/spaceship.png"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      id="entry_container"
    >
      <div
        id="entry_title_container"
      >
        <img
          alt="flight_horizons_title"
          id="entry_title_animated"
          src="https://flight-horizons.s3.us-west-2.amazonaws.com/title.png"
        />
      </div>
      <div
        id="entry_visual_container"
      >
        <img
          alt="spaceship_animation"
          id="entry_visual_animated"
          src="https://flight-horizons.s3.us-west-2.amazonaws.com/spaceship.png"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
